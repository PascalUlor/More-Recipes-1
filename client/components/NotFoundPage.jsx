import React, { Component } from 'react';
import Navbar from './NavBar.jsx';
import Footer from './Footer.jsx';


/**
 * @description defines 404 page not found component
 *
 * @class NotFoundPage
 *
 * @extends Component
 */
class NotFoundPage extends Component {
  /**
   * @description renders 404 page
   *
   * @returns { jsx } jsx - renders 404 page not found component
   */
  render() {
    return (
      <div className="bg-faded">
        <Navbar/>
        <main className="main-wrapper">
          <div className="not-found lead">
            <i
              className=
                "fa fa-exclamation-triangle fa-3x pb-3 d-block text-warning">
            </i>
            Ooops!!!.. Page Not Found
          </div>
        </main>
        <Footer/>
      </div>
    );
  }
}

export default NotFoundPage;
