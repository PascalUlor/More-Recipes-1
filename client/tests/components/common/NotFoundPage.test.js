/* eslint-disable */
import React from 'react';
import toJson from 'enzyme-to-json';
import NotFoundPage from '../../../components/NotFoundPage.jsx';


describe('<NotFoundPage/>', () => {
  it('renders 404(not found page) component without crashing', () => {
    const shallowWrapper = shallow(<NotFoundPage/>);
    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });
});
