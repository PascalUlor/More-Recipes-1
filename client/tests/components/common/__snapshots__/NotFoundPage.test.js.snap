// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage/> renders 404(not found page) component without crashing 1`] = `
<div
  className="bg-faded"
>
  <Connect(NavBar) />
  <main
    className="main-wrapper"
  >
    <div
      className="not-found lead"
    >
      <i
        className="fa fa-exclamation-triangle fa-3x pb-3 d-block text-warning"
      />
      Ooops!!!.. Page Not Found
    </div>
  </main>
  <Footer />
</div>
`;
